presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -883,7 +884,8 @@ private PlanBuilder sort(PlanBuilder subPlan, Optional<OrderBy> orderBy, List<Ex
                 new SortNode(
                         idAllocator.getNextId(),
                         subPlan.getRoot(),
-                        new OrderingScheme(orderBySymbols.build(), orderings)));
+                        new OrderingScheme(orderBySymbols.build(), orderings),
+                        isDistributedSortEnabled(session)));
```

Comment:
Split between partial/non partial sort happens in `io.prestosql.sql.planner.optimizations.AddExchanges.Rewriter#visitSort`